src/lsp_data.rs

```diff
@@ -85,11 +86,18 @@ pub mod ls_util {
         }
     }
 
-    pub fn range_from_vfs_file(_vfs: &Vfs, _fname: &Path) -> Range {
-        // FIXME: todo, endpos must be the end of the document, this is not correct
-
-        let end_pos = Position::new(0, 0);
-        Range{ start: Position::new(0, 0), end: end_pos }
+    pub fn range_from_vfs_file(_vfs: &Vfs, _fname: &Path) -> Result<Range, vfs::Error> {
```

Comment:
I think that it is more DRY to handle the error conversion as close to the source as possible, i.e., I don't want each client to have to repeat the error conversion.